File: admin-backend/admin-domain/src/main/java/fun/yozora/admin/domain/entity/Artifact.java
package fun.yozora.admin.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import java.io.Serial;
import java.io.Serializable;
import lombok.Data;

/**
 * 
 * @TableName artifact
 */
@TableName(value ="artifact")
@Data
public class Artifact implements Serializable {
    /**
     * 
     */
    @TableId(type = IdType.AUTO)
    private Integer id;

    /**
     * 
     */
    private String originId;

    /**
     * 
     */
    private String title;

    /**
     * 
     */
    private String url;

    /**
     * 
     */
    private String image;

    /**
     * 
     */
    private String location;

    /**
     * 
     */
    private String period;

    /**
     * 
     */
    private String material;

    /**
     * 
     */
    private String measurement;

    /**
     * 
     */
    private String artist;

    /**
     * 
     */
    private String creditLine;

    /**
     * 
     */
    private String type;

    /**
     * 
     */
    private String description;

    /**
     * 
     */
    private String illusion;

    @Serial
    @TableField(exist = false)
    private static final long serialVersionUID = 1L;
}

File: admin-frontend/apps/web-ele/src/types/artifact.ts
export interface Artifact {
  id: number;
  originId: string;
  title: string;
  url: string;
  image: string;
  location: string;
  period: string;
  material: string;
  measurement: string;
  artist: string;
  creditLine: string;
  type: string;
  description: string;
  illusion: string;
}
// #/types/artifact.ts

export interface Artifact {
  id: number;
  originId: string;
  title: string;
  url: string;
  image: string;
  location: string;
  period: string;
  material: string;
  measurement: string;
  artist: string;
  creditLine: string;
  type: string;
  description: string;
  illusion: string;
}

/**
 * 用于接口返回的数据结构，字段允许为 null
 */
export interface ArtifactItem {
  id: number;
  originId: string;
  title: string;
  url: string;
  image: string;
  location: null | string;
  period: null | string;
  material: null | string;
  measurement: null | string;
  artist: null | string;
  creditLine: null | string;
}
